text/0061-minimal-streams.md

```diff
@@ -0,0 +1,275 @@
+        - `valid: Out(1)`
+        - `ready: In(1)`
+    - with properties:
+        - `always_valid: bool`
+        - `always_ready: bool`
+    - with methods:
+        - `__eq__(other)`: requires `isinstance(other, amaranth.lib.stream.Signature)` and all three of `payload_shape`, `always_valid`, `always_ready` to match
+        - `create()`: constructs an `amaranth.lib.stream.Interface`
+- `amaranth.lib.stream.Interface(signature, *, path=None, src_loc_at=0)`
+    - with attributes:
+        - `.payload = Signal(payload_shape)`
+        - If `signature.always_valid == False`: `.valid = Signal()`
+        - If `signature.always_valid == True`:  `.valid = Const(1)`
+        - If `signature.always_ready == False`: `.ready = Signal()`
+        - If `signature.always_ready == True`:  `.ready = Const(1)`
+    - with properties:
+        - `p`: alias of `payload`
+
+Like other `amaranth.lib` modules, it is intended to be imported with the `from amaranth.lib import stream` statement and the classes be used with their qualified names, `stream.Signature` and `stream.Interface`.
+
+There is purposefully no `amaranth.lib.stream.Signature.payload_shape` property. It is expected that a future RFC will enable construction and introspection of complex streams, and the provision of any introspection of the payload is deferred until then.
+
+The `amaranth.lib.fifo.FIFOInterface` class is extended with properties:
+    - `r_stream: stream.Interface`: interface adapter with `payload = .r_data`, `valid = .r_rdy`, `ready = .r_en`.
+    - `w_stream: stream.Interface`: interface adapter with `payload = .w_data`, `valid = .w_en`, `ready = .r_rdy`.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+This proposal errs heavily on the side of minimalism, making it less broadly applicable.
+
+This proposal locks in two major design decisions:
+- That `payload` is a port member, not an interface member.
+- That the stream signature always has exactly these three members.
+
+If this turns out to be a mistake, existing code relying on the guarantees provided by these decisions will break.
+
+## Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+### A: Scope of this proposal
+[scope-of-this-proposal]: #scope-of-this-proposal
+
+Streams are one of the most highly coveted planned features. While we could choose to not provide them at all (as an external library implementation is possible), this will make Amaranth less competitive and perpetuate the status quo of every complex downstream project implementing their own, incompatible, version of streams.
+
+This module is described as "minimal" because it is highly limited in scope. This has the desirable qualities that it is easier to implement, document, and teach; that it is more widely compatible with existing code using streams; and that it is amenable to further gradual expansion.
+- The payload is an opaque (to the `amaranth.lib.stream` module) shape that is not given further meaning.
```

Comment:
I appreciate this decision. I was recently working on a codebase that tried to implement AXI-Stream using systemverilog interfaces (for synthesis) and found a number of bugs and data size issues caused by the interface containing unused fields. I think starting from the simplest possible channel is the right answer. (The discussion of litex streams below, also resonates)